ui/app/AppLayouts/Chat/ChatColumn/ChatMessages.qml

```diff
@@ -139,10 +139,12 @@ ScrollView {
                 chatLogView.scrollToBottom(true)
             }
 
-            onMessageNotificationPushed: function(chatId, msg, messageType, chatType, timestamp, identicon, username, hasMention) {
+            onMessageNotificationPushed: function(chatId, msg, messageType, chatType, timestamp, identicon, username, hasMention, isContact) {
                 if (appSettings.notificationSetting == Constants.notifyAllMessages || 
                     (appSettings.notificationSetting == Constants.notifyJustMentions && hasMention)) {
-                        notificationWindow.notifyUser(chatId, msg, messageType, chatType, timestamp, identicon, username)
+                        if (!chatType === Constants.chatTypeOneToOne || (appSettings.notifyNoneContactMessages || isContact)) {
```

Comment:
The condition here is a bit confusing. I guess at first you can remove the parenthesis since they don't do anything and move the `!` next to the `=`
```suggestion
                        if (chatType !== Constants.chatTypeOneToOne || appSettings.notifyNoneContactMessages || isContact) {
```
I feel like reversing the condition and exiting early would make it more straight forward:
```js
if (chatType === Constants.chatTypeOneToOne && !appSettings.notifyNoneContactMessages && !isContact)) {
  // We don't notify when one one on one chats from non-contacts when the setting is not activated
  return
}
```